src/osp/Resource/Package.cpp

```diff
@@ -32,9 +32,18 @@ namespace osp
 Package::Package(std::string const& prefix, std::string const& packageName) :
     // not sure about this but it compiles
     m_groups(),
-    //m_prefix{prefix[0], prefix[1], prefix[2], prefix[3]},
+    m_prefix(prefix),
     m_packageName(packageName)
 {
 }
 
+path_t decompose_path(std::string const& path)
```

Comment:
std::string_view. The caller of the function may not inherently have an std::string that they can pass in. They might have some other string type.

Reply:
This can be turned into a more generic string_view parsing function: inputting a string view and returning a pair of string views. If Path is intended to own a string value, then the pair can be used to initialize a Path.

Reply:
I can do that, although I'll probably end up making another `decompose_path` function that just wraps that one. The purpose of this one is to make it easy to parse resource names. I just realized that this is not used anywhere right now, but it will be once the rest of my branch is merged.